consensus/istanbul/backend/engine.go

```diff
@@ -777,6 +777,10 @@ func (sb *Backend) snapshot(chain consensus.ChainReader, number uint64, hash com
 		}
 
 		genesis := chain.GetHeaderByNumber(0)
+		if genesis == nil {
```

Comment:
When does this happen?  Seems like a critical issue if it does happen, and something we should panic on.

Reply:
It might happen when geth is exiting (using interrupt). Don't know what's the best way to have clean exit.

Reply:
OK.  Even then, that seems odd.  Since the genesis block should of been written to disk when `geth init` is done.

Reply:
agree with @kevjue. 

This should be a check we do on geth startup, in another context

Reply:
It's just messed up at some point when it's exiting (probably the DB has shutdown), clearly everything has been ok at startup.